theseus/embodied/robot/robot.py

```diff
@@ -51,11 +51,18 @@ def get_origin(urdf_origin: Optional[urdf.Pose] = None) -> torch.Tensor:
                 origin[:, :, 3] = torch.tensor(urdf_origin.xyz, dtype=dtype)
 
             if urdf_origin.rpy is not None:
-                rpy = urdf_origin.rpy
-                rot_x = so3.exp(torch.tensor([[rpy[0], 0, 0]], dtype=dtype))
-                rot_y = so3.exp(torch.tensor([[0, rpy[1], 0]], dtype=dtype))
-                rot_z = so3.exp(torch.tensor([[0, 0, rpy[2]]], dtype=dtype))
-                origin[:, :, :3] = rot_x @ rot_y @ rot_z
+                rpy = torch.tensor(urdf_origin.rpy, dtype=dtype)
```

Comment:
Was this rewritten for efficiency? (maybe to avoid matrix products?)

Reply:
That's correct.